packages/js-moi-manifest/src.ts/manifest.ts

```diff
@@ -187,35 +209,68 @@ export class ManifestCoder {
 
     /**
      * Encodes the provided arguments based on the given manifest routine 
-     * parmeters and its types (the accepts property in routine).
+     * parameters and its types (the accepts property in routine).
      * 
      * The arguments are mapped to their corresponding fields, and the calldata 
      * is generated by parsing and encoding the arguments based on the dynamically 
      * created schema from fields.
      *
-     * @param {LogicManifest.TypeField[]} fields - The fields associated with the 
-     routine parameters (arguments).
+     * @param {LogicManifest.TypeField[]} fields - The fields associated with the routine parameters (arguments).
+     * @param {any[]} args - The arguments to encode.
+     * @returns {string} The POLO-encoded calldata as a hexadecimal string prefixed with "0x".
+     */
+    public encodeArguments(fields: LogicManifest.TypeField[], args: any[]): string;
+    /**
+     * Encodes the provided arguments based on the given manifest routine
+     * parameters and its types (the accepts property in routine).
+     * 
+     * The arguments are mapped to their corresponding fields, and the calldata
+     * is generated by parsing and encoding the arguments based on the dynamically
+     * created schema from fields.
+     * 
+     * @param {string} fields - The name of the routine associated with the arguments.
+     * @param {any[]} args - The arguments to encode.
+     * @returns {string} The POLO-encoded calldata as a hexadecimal string prefixed with "0x".
+     */
+    public encodeArguments(fields: string, args: any[]): string;
+    /**
+     * Encodes the provided arguments based on the given manifest routine
+     * parameters and its types (the accepts property in routine).
+     * 
+     * The arguments are mapped to their corresponding fields, and the calldata
+     * is generated by parsing and encoding the arguments based on the dynamically
+     * created schema from fields.
+     * 
+     * @param {(LogicManifest.TypeField[] | string)} fields - The fields associated with the routine parameters (arguments) or the name of the routine.
      * @param {any[]} args - The arguments to encode.
-     * @returns {string} The POLO-encoded calldata as a hexadecimal string 
-     prefixed with "0x".
+     * @returns {string} The POLO-encoded calldata as a hexadecimal string prefixed with "0x".
      */
-    public encodeArguments(fields: LogicManifest.TypeField[], args: any[]): string {
+    public encodeArguments(fields: LogicManifest.TypeField[] | string, args: any[]): string {
+        if (typeof fields === "string") {
+            const element  = this.elementDescriptor.getRoutineElement(fields).data as LogicManifest.Routine
+            fields = element.accepts
+        }
+        
```

Comment:
Can we make it accept string alone.

Reply:
As we mentioned above `encodeArgument` take input as routine object or callsite name.